guix-science-nonfree/packages/nvidia-container-toolkit.scm

```diff
@@ -0,0 +1,203 @@
+    (home-page "https://github.com/NVIDIA/nvidia-modprobe")
+    (license gpl2)))
+
+(define-public libnvidia-container
+  (package
+    (name "libnvidia-container")
+    (version "1.13.1")
+    (source (origin
+              (method git-fetch)
+              (uri (git-reference
+                    (url "https://github.com/NVIDIA/libnvidia-container")
+                    (commit (string-append "v" version))))
+              (file-name (git-file-name name version))
+      	      (patches (search-patches "libnvidia-container.patch"))
+              (sha256
+               (base32
+		"0rzvh1zhh8pi5xjzaq3nmyzpcvjy41gq8w36dp1ai11a6j2lpa99"))))
+    (build-system gnu-build-system)
+    (arguments
+     (list
+      #:phases
+      #~(modify-phases %standard-phases
+	  (delete 'configure)
+	  (delete 'build)
+	  (delete 'check)
+          (add-after 'unpack 'ensure-writable-source
+	    (lambda* (#:key inputs outputs #:allow-other-keys)
+      	      (setenv "HOME" "/tmp")
+	      (make-file-writable "src/ldcache.c")
+      	      (make-file-writable "src/ldcache.h")
+              (make-file-writable "src/nvc_info.c")))
+	  (add-after 'patch-source-shebangs 'replace-prefix
+	    (lambda* (#:key inputs outputs #:allow-other-keys)
+	      (substitute* "Makefile"
+		(("/usr/local") (assoc-ref outputs "out"))  ;this overrides the prefix
+		(("debug??libdir?") "debug")  ;ensure debug files get installed in the correct subdir
+		((".*nvidia-modprobe.mk.*") "\n")
+		(("^all: shared static tools")
+		 "all: shared tools")
+		((".*LIB_STATIC.*libdir.*$") ""))
+     	      (substitute* "mk/nvcgo.mk"
+		((".*-rf.*")
+		 "\tmkdir -p ${SRCS_DIR} && echo \"sources dir: ${SRCS_DIR}\"\n")
+		(("CURDIR./src/..PREFIX.")
+		 "CURDIR)/src/$(PREFIX)/*")) ;deleting sources fails
+              (substitute* "src/cli/libnvc.c"
+		(("libnvidia-ml.so.1")
+		 "/run/current-system/profile/lib/libnvidia-ml.so.1"))
+              (substitute* "src/nvc_internal.h"
+		(("libnvidia-ml.so.1")
+		 "/run/current-system/profile/lib/libnvidia-ml.so.1"))
+	      (setenv "C_INCLUDE_PATH" (string-append (getenv "C_INCLUDE_PATH") ":" (string-append #$libtirpc "/include/tirpc")))
+     	      (setenv "LIBRARY_PATH" (string-append (getenv "LIBRARY_PATH") ":" (string-append #$libtirpc "/lib")))
+              (setenv "LDFLAGS" (string-append (or (getenv "LDFLAGS") "") " -ltirpc -lseccomp -lcap -Wl,-rpath=" (assoc-ref outputs "out") "/lib"))
+              (setenv "CFLAGS" (string-append (or (getenv "CFLAGS") "") " -DWITH_TIRPC -g"))
+       	      (substitute* "Makefile"
+		(("^WITH_LIBELF.*no")
+		 "WITH_LIBELF ?= yes"))
+    	      (substitute* "mk/common.mk"
+		(("^REVISION.*")
+		 (string-append "REVISION ?= " #$version "\n" "CC := gcc\n")))
+	      #true) ; must return true for success
+	    ))
+      #:tests? #f))
+    (native-inputs
+     (list libseccomp nvidia-modprobe which libtirpc libcap libelf git-minimal curl tar coreutils docker go gcc-toolchain rpcsvc-proto pkgconf))
```

Comment:
Since `gnu-build-system` is used, you can remove `tar`, `coreutils`, and `gcc-toolchain`: they're included already.

Reply:
done